### PR TITLE
polish llama 3 setup

### PR DESCRIPTION
**Pull request by tianyu-l**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #257

1. Use realist training configs on 64 A100 gpus.

2. Confirmed that `blobfile` pkg is needed (torchtune also has it).

3. The tokenizer download folder was not quite right. Revert the download destination of llama3 tokenizer to `torchtitan/datasets/tokenizer/original/tokenizer.model` as a temporary fix. According to HF [doc](https://huggingface.co/docs/huggingface_hub/en/guides/download):
> The downloaded files will keep the same file structure as in the repo. For example if filename="data/train.csv" and local_dir="path/to/folder", then the returned filepath will be "path/to/folder/data/train.csv".

So if we set local dir to be `torchtitan/datasets/tokenizer/llama3`, the file will be at `torchtitan/datasets/tokenizer/llama3/original/tokenizer.model`. Currently the fix in this PR aligns with README instructions. Let's discuss what's the best place to put them.

4. other minor fixes

### CODE REVIEW

File: train_configs/llama3_8b.toml [hunk 1]
```diff
@@ -16,15 +18,15 @@ save_tb_folder = "tb"
...
 flavor = "8B"
-norm_type = "fused_rmsnorm"  # layernorm / np_layernorm / rmsnorm / fused_rmsnorm
-tokenizer_path = "./torchtitan/datasets/tokenizer/llama3/tokenizer.model"
+norm_type = "fused_rmsnorm"  # [layernorm / np_layernorm / rmsnorm / fused_rmsnorm]
+tokenizer_path = "./torchtitan/datasets/tokenizer/original/tokenizer.model"
 
 [optimizer]
 name = "AdamW"
-lr = 3e-4
+lr = 1e-4
```
Comment (wanchaol) [inline train_configs/llama3_8b.toml]: why change to a smaller lr here?

File: train_configs/llama3_70b.toml [hunk 2]
```diff
@@ -19,14 +19,14 @@ save_tb_folder = "tb"
 name = "llama3"
 flavor = "70B"
 norm_type = "rmsnorm"  # [layernorm, np_layernorm, rmsnorm, fused_rmsnorm]
-tokenizer_path = "./torchtitan/datasets/tokenizer/llama3/tokenizer.model"
+tokenizer_path = "./torchtitan/datasets/tokenizer/original/tokenizer.model"
 
 [optimizer]
 name = "AdamW"
-lr = 1.5e-4
+lr = 5e-5
```
Comment (wanchaol) [inline train_configs/llama3_70b.toml]: ditto: why changing to a smaller lr? I think `1.5e-4 or 2e-4` is more realistic?

File: train_configs/llama3_70b.toml [hunk 3]
```diff
@@ -19,14 +19,14 @@ save_tb_folder = "tb"
...
 
 [training]
-batch_size = 16
+batch_size = 8
```
Comment (wanchaol) [inline train_configs/llama3_70b.toml:29]: why changing this to lower batch size?

Comment (tianyu-l) [reply train_configs/llama3_70b.toml:29]: global batch size 128 will cause oom when 1D + full AC is used for 70B
job link: https://www.internalfb.com/mlhub/pipelines/runs/mast/mfu-llama3-70b-1d-lbz2-cbtzw6

File: train_configs/llama3_8b.toml (hunk 1 above)
Comment (tianyu-l) [reply train_configs/llama3_8b.toml]: Before the change, learning rate will cause instability in training 
image

After the change, learning is slower but has nicer curves
image

In general, LLaMa3 seems "harder" to train. We can discuss which makes more sense to put as default.

Comment (wanchaol) [reply train_configs/llama3_8b.toml]: Ok I see. Instead of tuning the hyper params immediately, shall we land the rest changes first and make the hyperparam change be a separate change once we figure out the details? i.e. maybe we can adjust the learning rate scheduler instead of changing the learning rate?

Comment (tianyu-l) [reply train_configs/llama3_8b.toml]: sounds good, changed back
